packages/gestalt/src/Button.js

```diff
@@ -142,7 +161,108 @@ const ButtonWithForwardRef: React$AbstractComponent<
     </Text>
   );
 
+  const iconEndComponent = (): Node => {
+    const SIZE_NAME_TO_PIXEL = {
+      sm: 10,
+      md: 12,
+      lg: 12,
+    };
+
+    return (
+      <Box alignItems="center" display="flex">
+        {buttonText}
+        <Box display="inlineBlock" flex="none" marginStart={2}>
+          <Icon
+            accessibilityLabel=""
+            color={textColor}
+            icon={iconEnd}
+            size={SIZE_NAME_TO_PIXEL[size]}
+          />
+        </Box>
+      </Box>
+    );
+  };
+
+  function handleClick(
+    event:
+      | SyntheticMouseEvent<HTMLButtonElement>
+      | SyntheticKeyboardEvent<HTMLButtonElement>
+  ): void {
+    if (onClick) {
+      onClick({ event });
+    }
+  }
+
+  function handleLinkClick({
+    event,
+  }: {|
+    event:
+      | SyntheticMouseEvent<HTMLAnchorElement>
+      | SyntheticKeyboardEvent<HTMLAnchorElement>,
+  |}): void {
+    if (onClick) {
+      onClick({ event });
+    }
+  }
+
   /* eslint-disable react/button-has-type */
+  if (props.role === 'link') {
+    const { href, rel, target } = props;
+
+    return (
+      <Link
+        accessibilityLabel={accessibilityLabel}
+        _disabled={disabled}
```

Comment:
@AlbertCarreras Let's not make this a pattern and add `_` props for "private" props. People can still use them.

Instead, let's add `disabled` to `Link`

Reply:
DONE